### ISSUE TITLE
Add Support for Kubernetes Service Catalog

### ISSUE THREAD

**Issue by salaboy** [spring-cloud/spring-cloud-kubernetes#231]
We should support the interaction against the Kubernetes Service Catalog, @iocanel has an implementation already that we can test and align with this projects: http://iocanel.com/2018/09/service-catalog-pt-2/

**Comment by geoand** [comment]
You probably won't be surprised @Salaboy  that I'm interested in hearing your ideas about this as well :)

**Comment by MichalFoksa** [comment]
I vote for it if it means having access to kind of data from `io.fabric8.kubernetes.api.model.Service` and 
`io.fabric8.kubernetes.api.model.ServiceSpec`.

**Comment by eddie71** [comment]
Would this not be a separate component to support the spring cloud connectors projects?

**Comment by spencergibb** [comment]
We have no plans to update spring cloud connectors

**Comment by spencergibb** [comment]
If anything it might end up looking like https://github.com/pivotal-cf/java-cfenv

**Comment by spencergibb** [comment]
See https://github.com/buildpack/rfcs/blob/master/text/0012-service-binding.md#structure for potential structure.

**Comment by wind57** [comment]
@ryanjbaxter isn't this something you already added?

**Comment by ryanjbaxter** [comment]
Nope, but there is already this https://github.com/spring-cloud/spring-cloud-bindings

**Comment by wind57** [comment]
we should close this ticket, imo. 

The title says : "Add Support for Kubernetes Service Catalog", but support for that has been sunsetted : https://github.com/kubernetes-retired/service-catalog 

@ryanjbaxter wdyt?

**Comment by salaboy** [comment]
Closing is fine
